API.Tests/Integration/SpaceTests.cs

```diff
@@ -0,0 +1,331 @@
+        var apiSpace = await response.ReadAsHydraResponseAsync<Space>();
+        
+        // assert
+        apiSpace.Name.Should().Be("Patch Space Before");
+        
+        const string patchJson = @"{
+  ""@type"": ""Space"",
+  ""name"": ""Patch Space After""
+}";
+        var patchContent = new StringContent(patchJson, Encoding.UTF8, "application/json");
+        var patchResponse = await httpClient.AsCustomer(customerId.Value).PatchAsync(apiSpace.Id, patchContent);
+        var patchedSpace = await patchResponse.ReadAsHydraResponseAsync<Space>();
+
+        patchedSpace.Name.Should().Be("Patch Space After");
+    }
+    
+    [Fact]
+    public async Task Patch_Space_Leaves_Omitted_Fields_Intact()
+    {
+        // arrange
+        int? customerId = await EnsureCustomerForSpaceTests("Patch_Space_Leaves_Omitted_Fields_Intact");
+        
+        const string newSpaceJson = @"{
+          ""@type"": ""Space"",
+          ""name"": ""Patch Complex Space"",
+          ""defaultRoles"": [""role1"", ""role2""],
+          ""defaultTags"": [""tag1"", ""tag2""],
+          ""maxUnauthorised"": 400
+        }";
+        const string patchJson = @"{
+          ""@type"": ""Space"",
+          ""name"": ""Patch Complex Space After""
+        }";
+        
+        // act
+        var content = new StringContent(newSpaceJson, Encoding.UTF8, "application/json");
+        var postUrl = $"/customers/{customerId}/spaces";
+        var response = await httpClient.AsCustomer(customerId.Value).PostAsync(postUrl, content);
+        var apiSpace = await response.ReadAsHydraResponseAsync<Space>();
+        
+        // assert
+        var patchContent = new StringContent(patchJson, Encoding.UTF8, "application/json");
+        var patchResponse = await httpClient.AsCustomer(customerId.Value).PatchAsync(apiSpace.Id, patchContent);
+        var patchedSpace = await patchResponse.ReadAsHydraResponseAsync<Space>();
+
+        patchedSpace.Name.Should().Be("Patch Complex Space After");
+        patchedSpace.DefaultRoles.Should().BeEquivalentTo("role1", "role2");
+        patchedSpace.DefaultTags.Should().BeEquivalentTo("tag1", "tag2");
+        patchedSpace.MaxUnauthorised.Should().Be(400);
+    }
+    
+    private async Task<int?> EnsureCustomerForSpaceTests(string customerName = "space-test-customer")
+    {
+        var spaceTestCustomer = await dbContext.Customers.SingleOrDefaultAsync(c => c.Name == customerName);
+
+        if (spaceTestCustomer != null)
+        {
+            return spaceTestCustomer.Id;
+        }
+        
+        string spaceTestCustomerJson = $@"{{
+  ""@type"": ""Customer"",
+  ""name"": ""{customerName}"",
+  ""displayName"": ""Display - {customerName}""
+}}";
+        
+        var content = new StringContent(spaceTestCustomerJson, Encoding.UTF8, "application/json");
+        var response = await httpClient.AsAdmin().PostAsync("/customers", content);
+        var apiCustomer = await response.ReadAsHydraResponseAsync<Customer>();
```

Comment:
Is it necessary to create customer via API rather than DB? Feels like we're making these tests more brittle. A change in `POST /customers` behaviour (e.g. a new, required field) could end up breaking some of these `Space` tests; when in reality all of the `Space` endpoints are fine.

The benefit is that `POST /customers` could do something more than create a customer that is necessary for creating spaces but that 'more' behaviour should be asserted in customer tests.